emcc.py

```diff
@@ -573,16 +572,10 @@ def uniquename(name):
   else:
     final_suffix = ''
 
-  if TEMP_DIR:
-    temp_dir = TEMP_DIR
-    if os.path.exists(temp_dir):
-      shutil.rmtree(temp_dir) # clear it
-    os.makedirs(temp_dir)
-  else:
-    temp_root = shared.TEMP_DIR
-    if not os.path.exists(temp_root):
-      os.makedirs(temp_root)
-    temp_dir = tempfile.mkdtemp(dir=temp_root)
+  temp_root = shared.TEMP_DIR
+  if not os.path.exists(temp_root):
```

Comment:
creating `shared.TEMP_DIR` could be done in `shared`, I think? seems cleaner that way, unless I'm missing something, and we don't want `shared` to always create it?

Reply:
Some tools including test runners, emar, emconfigure, etc. also imports `shared`, we probably don't want to create and remove empty directory for that tools.

Reply:
Good point, yes.